packages/eslint-plugin-template/src/rules/no-duplicate-attributes.ts

```diff
@@ -0,0 +1,89 @@
+import { TmplAstElement } from '@angular/compiler';
+import {
+  createESLintRule,
+  getTemplateParserServices,
+} from '../utils/create-eslint-rule';
+import {
+  BoundAttribute,
+  BoundEvent,
+  TextAttribute,
+} from '@angular/compiler/src/render3/r3_ast';
+
+type Options = [];
+export type MessageIds = 'noDuplicateAttributes';
+export const RULE_NAME = 'no-duplicate-attributes';
+
+export default createESLintRule<Options, MessageIds>({
+  name: RULE_NAME,
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description:
+        'Ensures that no components have duplicate input properties or output event listeners',
+      category: 'Best Practices',
+      recommended: false,
+    },
+    schema: [],
+    messages: {
+      noDuplicateAttributes:
+        'The attribute "{{attributeName}}" was declared multiple times',
+    },
+  },
+  defaultOptions: [],
+  create(context) {
+    const parserServices = getTemplateParserServices(context);
+
+    return {
+      Element({ inputs, outputs, attributes }: TmplAstElement) {
+        const duplicateInputsAndAttributes = findDuplicates([
+          ...inputs,
+          ...attributes,
+        ]);
+
+        const nonTwoWayBindingOutputs = outputs.filter((output) => {
+          return !inputs.find(
+            (input) =>
+              input.sourceSpan.start === output.sourceSpan.start &&
+              input.sourceSpan.end === output.sourceSpan.end,
+          );
+        });
+
+        const duplicateOutputs = findDuplicates(nonTwoWayBindingOutputs);
+
+        const allDuplicates = [
+          ...duplicateInputsAndAttributes,
+          ...duplicateOutputs,
+        ];
+
+        allDuplicates.forEach((duplicate) => {
+          const loc = parserServices.convertNodeSourceSpanToLoc(
+            duplicate.sourceSpan,
+          );
+
+          context.report({
```

Comment:
This will output 2 errors for each duplicate attribute e.g. `<input [name]="foo" [name]="bar">` would output 2 errors, one for `[name]="foo"` and one for `[name]="bar"`. I think this would be the expected behaviour but it would be good to get confirmation that this is correct.

Reply:
By reporting the second attribute as an error though I think it might make developers introduce bugs when fixing the issue. e.g. if this was reported to me as an error:
```
<input [name]="foo" [name]="bar">
                    ~~~~~~~~~~~~
```

My instinct would be just to follow the error and change the code to:
```
<input [name]="foo">
```

However this would now introduce a bug as the value of `[name]` would have changed from `bar` to `foo`. Really I should have deleted the first attribute.

and just to make the situation more complex, if you're not using ivy then it's the second attribute that should have been removed 😅

Reply:
My main concern is the visual noise and possibly unnecessary reports 😞 , but now I understood why you went with this option.

Reply:
Yeah tricky one - let's stick with flagging both for now and let the user make the decision.